test/generic/MPoly-test.jl

```diff
@@ -113,18 +113,48 @@
    @test y isa Generic.MPoly{Generic.Poly{BigInt}}
    @test z isa Generic.MPoly{Generic.Poly{BigInt}}
 
-   ZZxyz, (x,y,z) = polynomial_ring(ZZ, 'x':'z')
-   @test ZZxyz isa Generic.MPolyRing
+   ZZxyz_ = polynomial_ring(ZZ, 'x':'z')
+   ZZxyz2, xyz2... = polynomial_ring(ZZ, (:x, 'y', GenericString("z")))
+   ZZxyz3, xyz3... = polynomial_ring(ZZ, :x, 'y', GenericString("z"))
+   ZZxyz4_ = polynomial_ring(ZZ, Union{String,Char,Symbol}["x", 'y', :z])
+   ZZxyz5_ = ZZ["x", 'y', :z]
+   ZZxyz6 = @polynomial_ring(ZZ, :x, :y, :z)
 
-   ZZxyz2, (x2,y2,z2) = polynomial_ring(ZZ, (:x, 'y', GenericString("z")))
-   @test ZZxyz == ZZxyz2
-   @test (x,y,z) == (x2,y2,z2)
+   @test ZZxyz_[1] isa Generic.MPolyRing
+   @test ZZxyz_ == (ZZxyz2, collect(xyz2))
+   @test ZZxyz_ == (ZZxyz3, collect(xyz3))
+   @test ZZxyz_ == ZZxyz4_
+   @test ZZxyz_ == ZZxyz5_
+   @test ZZxyz_ == (ZZxyz6, [x, y, z])
 
-   ZZxyz3, _ = polynomial_ring(ZZ, Union{String,Char,Symbol}["x", 'y', :z])
-   @test ZZxyz == ZZxyz3
+   ZZxxx0_ = polynomial_ring(ZZ, :x=>Base.OneTo(3))
+   ZZxxx_ = polynomial_ring(ZZ, :x=>1:3)
+   ZZxxx2, (xxx2,) = ZZ[:x=>1:3]
+   ZZxxx3_ = polynomial_ring(ZZ, "x[@]"=>1:3)
 
-   ZZxyz4, _ = ZZ["x", 'y', :z]
-   @test ZZxyz == ZZxyz4
+   @test ZZxxx_[1] isa Generic.MPolyRing
+   @test ZZxxx_ == ZZxxx0_
+   @test ZZxxx_ == (ZZxxx2, xxx2)
+
+   QQxxx_ = polynomial_ring(QQ, "x#" => 1:3)
+   QQxxx2 = @polynomial_ring(QQ, "x#" => 1:3)
+
+   @test QQxxx_[1] isa Generic.MPolyRing
+   @test QQxxx_ == (QQxxx2, [x1, x2, x3])
+
+   ZZxy_ = polynomial_ring(ZZ, :x => (1:2, 1:2), :y => 0:3)
+   ZZxy2_ = polynomial_ring(ZZ, :x => ["1,1" "1,2"; "2,1" "2,2"], :y => (0:3,))
```

Comment:
Is there a difference between the effect of `:y => 0:3` and `:y => (0:3,)`? Or are these just here to test two different variants (which is of course fine)

Reply:
The latter, i.e. the test verifies that `:y => 0:3` and `:y => (0:3,)` yield the same result.